_posts/ja/newsletters/2024-11-29-newsletter.md

```diff
@@ -0,0 +1,198 @@
+---
+title: 'Bitcoin Optech Newsletter #331'
+permalink: /ja/newsletters/2024/11/29/
+name: 2024-11-29-newsletter-ja
+slug: 2024-11-29-newsletter-ja
+type: newsletter
+layout: newsletter
+lang: ja
+---
+今週のニュースレターでは、Bitcoinスクリプト用のLisp方言に関する最近の議論を掲載しています。
+また、Bitcoin Stack Exchangeで人気の質問とその回答や、新しいリリースとリリース候補の発表、
+人気のあるBitcoinインフラストラクチャプロジェクトの注目すべき変更など恒例のセクションも含まれています。
+
+## ニュース
+
+- **Bitcoinスクリプト用のLisp方言:** Anthony Townsは、
+  ソフトフォークでBitcoinに追加可能なBitcoin用のLisp方言を作成する[研究][topic bll]の続きについて、
+  いくつか投稿しました。
+
+  - *bll, symbll, bllsh:* Townsは、Chia Lisp開発者のArt Yerkesからの、
+    （プログラマーが通常書く）高レベルコードと（実際に実行されるもので、
+    通常はコンパイラによって高レベルコードから作成される）低レベルコード間の適切なマッピングを確保するためのアドバイスについて、
+    長い間考えていたと[述べています][towns bllsh1]。彼は、
+    「（miniscriptがスクリプトに対して行うように）高レベル言語を低レベル言語の使いやすいバリエーションとして扱う」
+    という[miniscript][topic miniscript]のようなアプローチを取ることにしました。
+    その結果、2つの言語と1つのツールができました。
+
+    - *Basic Bitcoin Lisp言語 (bll)* は、ソフトフォークでBitcoinに追加できる低レベルの言語です。
+      Townsによると、前回の更新時点（[ニュースレター #294][news294 btclisp]参照）で、
+      bllはBTC Lispに似ています。
+
+    - *シンボリックbll (symbll)* は、bllに変換される高レベル言語です。
+      関数型プログラミングに慣れている人にとっては、比較的簡単なはずです。
+
+    - *Bllシェル (bllsh)* は、bllとsymbllでスクリプトをテストし、
+      symbllからbllにコンパイルし、デバッグ機能を使用してコードを実行できる[REPL][]です。
+
+  - *symbllとGSRにおける量子安全な署名の実装:*
+    Townsは、既存のopcodeとRusty RussellのGSR（_Great Script Restoration_）[提案][russell gsr]で定義されたopcodeを使用して
+    Winternitz One Time Signatures (WOTS+)を実装することに関するJonas Nickの[Twitter投稿][nick wots]を[リンク][towns wots]しています。
+    Townsは、次にbllshでsymbllを使用してWOTSを実装することを比較しました。
+    これにより、オンチェーンに配置する必要があるデータ量が少なくとも83%、場合によっては95%以上削減されます。
+    これにより、P2WPKHアウトプットの30倍のコストで[量子安全な署名][topic quantum resistance]を使用することができます。
+
+  - *<!--flexible-coin-earmarks-->柔軟なコインの用途指定:*
+    Townsは、１つのUTXOを特定の金額と使用条件に分割できるsymbll（およびおそらく[Simplicity][topic simplicity]）と互換性のある
+    汎用構造について[説明しています][towns earmarks]。使用条件が満たされると、
+    関連付けられた金額を使用することができ、UTXOの残りの金額は残りの条件と一緒に新しいUTXOに戻されます。
+    UTXOのすべて使用できるようにする別の条件が満たされる場合もあります。たとえば、
+    これによりすべての参加者が条件の一部を更新することに同意できるようになります。これはTownsが以前提案した
+    `OP_TAP_LEAF_UPDATE_VERIFY`（TLUV、[ニュースレター #166][news166 tluv]参照）に似た
+    柔軟なタイプの[コベナンツ][topic covenants]の仕組みですが、Townsは以前、
+    コベナンツは「正確でも有用な用語でもない」と考えていると[書いています][towns badcov]。
+
+    （[LN-Symmetry][topic eltoo]ベースのチャネルを含む）LNチャネルのセキュリティとユーザビリティの向上や、
```

Comment:
Is this newline intentional?

Reply:
This line break corresponds to the following blank line in the original:

```
    badcov] that he thinks _covenants_ is "not an accurate or useful
    term".

    Several examples for how these _flexible coin earmarks_ can be used
    are provided, including improvements in the security and usability
```